lib/iris/coords.py

```diff
@@ -732,7 +732,13 @@ def __str__(self):
             fmt = '{cls}({points}{bounds}' \
                   ', standard_name={self.standard_name!r}' \
                   ', calendar={self.units.calendar!r}{other_metadata})'
-            points = self._str_dates(self.points)
+            if self.units.is_long_time_interval():
```

Comment:
This makes Iris have a hard dependency on cf_units >= 1.2. We need to update the requirements to reflect that (or not have such a hard dependency)